src/main/java/JsonIO.java

```diff
@@ -30,15 +33,75 @@ void writeToJson(String pathToJson, Object object) {
 
     Snippet loadSnippetFromJson(String pathToJson) {
         try {
-            return g.fromJson(new FileReader(pathToJson), Snippet.class);
+            return gson.fromJson(new FileReader(pathToJson), Snippet.class);
         } catch (FileNotFoundException ex) {
             onException(ex);
             return null;
         }
     }
 
+    List<Snippet> parseSnippetsFromGistsApi(String allSnippetsData) {
+        List<Snippet> parsedSnippets = new ArrayList<Snippet>();
+
+        JsonArray gistArray = gson.fromJson(allSnippetsData, JsonArray.class);
+
+        for (int i = 0; i < gistArray.size(); i++) {
+            String gistUrl = gistArray.get(i).getAsJsonObject().get("id").getAsString();
+            Snippet fetchedSnippet = GistsApi.getInstance().getSpecificSnippet(gistUrl);
+            parsedSnippets.add(fetchedSnippet);
+        }
+
+        return parsedSnippets;
+    }
+
+    Snippet parseSnippetAttributesFromGistsApi(String snippetData) {
+        JsonObject completeGistObject = gson.fromJson(snippetData, JsonObject.class);
+        JsonObject relevantGistData = completeGistObject
```

Comment:
A lot of functions used on one object, but solved in a nice way by splitting it up and adding comments. Clear!